src/world.rs

```diff
@@ -1360,4 +1371,68 @@ mod tests {
         let mut world = World::new();
         assert!(world.insert_one(Entity::DANGLING, ()).is_err());
     }
+
+    #[test]
+    fn view_borrowed() {
+        let mut world = World::new();
+        let e0 = world.spawn((3, "hello"));
+        let e1 = world.spawn((6.0, "world"));
+        let e2 = world.spawn((12,));
+
+        {
+            let str_view = world.view::<&&str>();
+
+            assert_eq!(*str_view.get(e0).unwrap(), "hello");
+            assert_eq!(*str_view.get(e1).unwrap(), "world");
+            assert_eq!(str_view.get(e2), None);
+        }
+
+        {
+            let mut int_view = world.view::<&mut i32>();
+            assert_eq!(*int_view.get_mut(e0).unwrap(), 3);
+            assert_eq!(int_view.get_mut(e1), None);
+            assert_eq!(*int_view.get_mut(e2).unwrap(), 12);
+
+            // edit some value
+            *int_view.get_mut(e0).unwrap() = 100;
+        }
+
+        {
+            let mut int_str_view = world.view::<(&&str, &mut i32)>();
+            let (s, i) = int_str_view.get_mut(e0).unwrap();
+            assert_eq!(*s, "hello");
+            assert_eq!(*i, 100);
+            assert_eq!(int_str_view.get_mut(e1), None);
+            assert_eq!(int_str_view.get_mut(e2), None);
+        }
+    }
+
+    #[test]
+    fn view_mut() {
+        let mut world = World::new();
+        let e0 = world.spawn((3, "hello"));
+        let e1 = world.spawn((6.0, "world"));
+        let e2 = world.spawn((12,));
+
+        let str_view = world.view_mut::<&&str>();
+
+        assert_eq!(*str_view.get(e0).unwrap(), "hello");
+        assert_eq!(*str_view.get(e1).unwrap(), "world");
+        assert_eq!(str_view.get(e2), None);
+
+        let mut int_view = world.view_mut::<&mut i32>();
+        assert_eq!(*int_view.get_mut(e0).unwrap(), 3);
+        assert_eq!(int_view.get_mut(e1), None);
+        assert_eq!(*int_view.get_mut(e2).unwrap(), 12);
+
+        // edit some value
+        *int_view.get_mut(e0).unwrap() = 100;
+
+        let mut int_str_view = world.view_mut::<(&&str, &mut i32)>();
+        let (s, i) = int_str_view.get_mut(e0).unwrap();
+        assert_eq!(*s, "hello");
+        assert_eq!(*i, 100);
+        assert_eq!(int_str_view.get_mut(e1), None);
+        assert_eq!(int_str_view.get_mut(e2), None);
+    }
```

Comment:
nit: the above are integration tests, not really particular to the internals of `World` specifically, and might hence make more sense in tests/tests.rs.